### ISSUE TITLE
Can't see 'system settings' section inside Configuration Center

### ISSUE THREAD

**Issue by fulalas** [lxde/lxqt#1307]
This is more like a question. I'm building a LXQt module for Porteus distro and I've noticed that I don't have the section 'system settings' inside Configuration Center, although this is present in many other LXQt distros usually with lxqt-admin-time, lxqt-admin-user and gparted.

**Edited**: I've figured it out. You have to edit each /usr/share/applications/*.desktop and add 'Settings' in the section Categories

**Comment by tsujan** [comment]
If by 'system settings' you mean "KDE System Settings" (systemsettings5), I've never seen it inside "Configuration Center", as it shouldn't be there.

**Comment by fulalas** [comment]
Take a look here: 
![manjaro-lxqt-16-03-lxqt-configuration-center](https://user-images.githubusercontent.com/27843666/26860741-b5846ad4-4b93-11e7-9f5e-4c89a3ea19e0.jpg)

In fact I was able to add GParted shortcut inside 'System settings' section. But 'Date And Time' and 'User And Groups' don't appear; only if I change Categories to =Settings;DesktopSettings;Qt;LXQt; But then it appears inside 'LXQt settings' section, which isn't the right place. Any suggestion?

**Comment by tsujan** [comment]
My mistake! You meant the section "System settings", not KDE's app with the same name.

> But then it appears inside 'LXQt settings' section, which isn't the right place. Any suggestion?

Change their category to something like this if you want them under "System settings":
```
Categories=Settings;System; 
```

**Comment by tsujan** [comment]
BTW, there's no reason for all config apps to appear on "Configuration Center". As you've found out, they should be in certain categories. If you find any inconsistency, feel free to open this issue!

**Comment by fulalas** [comment]
None of them appear :( They only appear inside main menu -> Preferences

**Comment by tsujan** [comment]
Do you mean that you changed their categories to `Categories=Settings;System; `, closed "Configuration Center", reopened it and they didn't show up under "System settings"? Here I can't reproduce this issue (if it exists). Maybe your installed version isn't up-to-date. Mine is the latest git.

**Comment by fulalas** [comment]
I just rebooted and everything is working now. Thanks, man! :)

**Comment by fulalas** [comment]
Sorry for bringing this back, but although 'Date and Time' now appears inside Configuration Center, if I change anything inside it, I receive this error:

![error](https://user-images.githubusercontent.com/27843666/27285848-2a801708-5552-11e7-995c-aa8fc7468c50.png)

Also, when executing lxqt-admin-user with lxqt-sudo, if you try to change current user password, you'll have to re-enter the new password 3 times through lxqt-sudo dialog, which isn't clear at all. Is this the expected behavior?

**Comment by tsujan** [comment]
I don't think this can be about LXQt.

**Comment by agaida** [comment]
hmm - a non-systemd config?

**Comment by fulalas** [comment]
@tsujan, is there any forum or any other channel where we can ask things about how LXQt works on the user/configuration level? I'm asking this because I'm having a few problems compiling and configuring LXQt. And this not only for my personal use, but for a community. So it would be really nice if I could find a place where people exchange their knowledge about LXQt. I think it would help spreading this wonderful -- and, at the moment, little known/used -- DE.

@agaida, what you mean exactly? What 'Date and Time' needs in order to change system date/time setting?

**Comment by tsujan** [comment]
@fulalas Which package provides "Date and Time"? LXQt should be compatible with all Qt5 apps.

**Comment by fulalas** [comment]
Sorry, but I didn't understand your question. 'Date and Time' is part of LXQt as 'lxqt-admin-time' bin. I'm using Qt 5.8.0.

**Comment by tsujan** [comment]
> Sorry, but I didn't understand your question

Oh, I thought "Date and Time" was a foreign app. Since I don't use GUI apps for administrative jobs, I shouldn't reply in this case. I apologize for that.

If you've found a bug in 'lxqt-admin`, please report it separately.

**Comment by fulalas** [comment]
It's probably not a bug since some distros have this working flawlessly. Maybe I'm missing some additional package in order to make it work. The question is: what 'lxqt-admin-time' (I think it's better calling this way :) ) depends on? 

Also, there's this other problem regarding lxqt-admin-user.

I can't see any issue section inside lxat-admin git page. Do I have to use main lxqt issue page?

**Comment by palinek** [comment]
> The question is: what 'lxqt-admin-time' (I think it's better calling this way :) ) depends on?

It uses the interface provided by [systemd/timedated](https://www.freedesktop.org/wiki/Software/systemd/timedated/) daemon.

**Comment by fulalas** [comment]
@palinek, thanks for the tip. I'll try to fix that. I would like to ask you something: why depend on systemd if we can change date/time natively?

**Comment by palinek** [comment]
> why depend on systemd if we can change date/time natively? Lots of Linux distro don't use systemd, right?

It was based on liboobs before, which is obsolete... see https://github.com/lxde/lxqt/issues/495 and https://github.com/lxde/lxqt-admin/pull/34.

If you know about something more "universal", let us know...

**Comment by fulalas** [comment]
QProcess + https://www.cyberciti.biz/faq/howto-set-date-time-from-linux-command-prompt/

What you think?

**Comment by palinek** [comment]
Yes, you can set the current time with `date`, but how about getting/setting timezone(s)?

**Comment by fulalas** [comment]
Get current localtime -> /etc/localtime (it's a symlink pointing to /usr/share/zoneinfo/*selectedLocalTime*)

Set localtime -> create the symlink /etc/localtime pointing to /usr/share/zoneinfo/*selectedLocalTime*

I think it worths the effort since by doing that we're not creating a dependency anymore. :) Take a look at this list of distros that don't use systemd: http://without-systemd.org

**Comment by palinek** [comment]
So basically do the very same, that the timedated daemon does? https://github.com/systemd/systemd/blob/master/src/timedate/timedated.c

I'm not sure if it's worth when other parts of LXQt already depend on other systemd components.

**Comment by fulalas** [comment]
Personally I don't like the idea of having a dependency on something that is not trivial to install. One thing is depending of one or two simple libs. Other completely different is to depend on a huge and controversial project like systemd, a project that is being reject by so many people and distros. So, yes, sometimes I think it's better to re-write the wheel (or copy it, ha ha) if it extends compatibility.

BTW, what other parts of LXQt depends on systemd?

**Comment by Chiitoo** [comment]
> BTW, what other parts of LXQt depends on systemd?

I'm interested on this too (could not find anything over at the wiki here).

I started using LXQt not long ago, and ended up packaging it for Gentoo, but did not bump into such a dependency before seeing this.  If it becomes something LXQt can't live without, then I'll be unfortunately unable to package it (or use it) any longer.  :\

**Comment by agaida** [comment]
There are no real dependencies on systemd - thats all, nothing to see here. And for the timedate and user administration please have a deeper look into the code. (I didn't, but i know we get BSD patches for that) - so i guess that there are no hard dependencies on systemd.

**Comment by fulalas** [comment]
@agaida, you said: 'hmm - a non-systemd config?'. I didn't understand exactly what you meant. Then I made some research and found that 'org.freedesktop.timedate1' refers to systemd. Now you're saying that there's no dependency on systemd. It's a bit confusing for me. Could you elaborate? I just want to make it work and I don't want to install systemd. Thanks! :)

**Comment by agaida** [comment]
org.freedesktop.timedate1 reads like a freedesktop interface - so if one outside of systemd implement it it should just work (tm) - second thing is: one can use binaries from the systemd source tree without the need of systemd as PID1.

**Comment by fulalas** [comment]
I've tried to install just systemd libs and freedesktop interfaces, but when I try to change date/time I receive a generic error (something like: system returns with error 1), so I can't investigate it further. It seems there's no easy way to fix that without a true systemd installation. Do you have a more specific solution? Thanks!

**Comment by agaida** [comment]
the culprit is:
```
TimeDateCtl::TimeDateCtl()
{
    mIface = new QDBusInterface(QStringLiteral("org.freedesktop.timedate1"),
                                QStringLiteral("/org/freedesktop/timedate1"),
                                QStringLiteral("org.freedesktop.timedate1"),
                                QDBusConnection::systemBus());
}
```
so you will need something that provide this dbus interface

**Comment by fulalas** [comment]
@agaida, thanks for the tips. I tried, but there's no systemd-shim for Slackware, so I tried to adapt Ubuntu package. It turns out that it depends on cgmanager, and after installing it I got 2 warnings and a critical error regarding systemd. After fixing the critical error and finally running shim, lxqt-admin-time, I still get this error when trying to change date and/or time:

![lxqt-admin-time](https://user-images.githubusercontent.com/27843666/27515097-ed65d450-59ef-11e7-8aaf-53e7061ca37e.png)

I'm wondering if this endless systemd issue is more reasonable than simply get rid of it on LXQt, since changing date/time manually is not that hard. Although I'm not an expert on Linux/LXQt/systemd, I know I'm above the medium user, and even considering that I can't find a solution. Imagine that as me and **Chiitoo**, others also could give up putting LXQt inside some distros because of this. And it would be sad because LXQt is really amazing in almost every aspect.
